### PR TITLE
Improved image gradient GetValue(s) performance by using cached image data

### PR DESCRIPTION
**Pull request by cgalvan**
In the original change of the `ImageGradient` to use a `StreamingImageAsset` instead of the custom `GradientSignal::ImageAsset`, the GetValue(s) benchmarks performance degraded. This is because previously the `GradientSignal::ImageAsset` owned the raw byte data itself, as opposed to the `StreamingImageAsset` which has to go through the mip chain to retrieve the actual sub-image data. To optimize the performance, I've updated the `ImageGradient` to cache the raw image data once the asset has been loaded. This uses a new companion API that retrieves pixel values directly from a vector of bytes instead of passing a `StreamingImageAsset`.

GradientSignal ImageGradient benchmarks before change:
```
GradientGetValues/BM_ImageGradient/EbusGetValue:0/size:1024                      58.1 ms         57.8 ms           10
GradientGetValues/BM_ImageGradient/EbusGetValue:0/size:2048                       236 ms          240 ms            3
GradientGetValues/BM_ImageGradient/EbusGetValue:0/size:4096                       933 ms          938 ms            1
GradientGetValues/BM_ImageGradient/EbusGetValues:1/size:1024                     35.7 ms         36.2 ms           19
GradientGetValues/BM_ImageGradient/EbusGetValues:1/size:2048                      144 ms          145 ms            4
GradientGetValues/BM_ImageGradient/EbusGetValues:1/size:4096                      575 ms          578 ms            1
GradientGetValues/BM_ImageGradient/SamplerGetValue:2/size:1024                   77.1 ms         76.4 ms            9
GradientGetValues/BM_ImageGradient/SamplerGetValue:2/size:2048                    315 ms          312 ms            2
GradientGetValues/BM_ImageGradient/SamplerGetValue:2/size:4096                   1253 ms         1250 ms            1
GradientGetValues/BM_ImageGradient/SamplerGetValues:3/size:1024                  37.0 ms         37.0 ms           19
GradientGetValues/BM_ImageGradient/SamplerGetValues:3/size:2048                   148 ms          148 ms            4
GradientGetValues/BM_ImageGradient/SamplerGetValues:3/size:4096                   593 ms          594 ms            1
```

GradientSignal ImageGradient benchmarks after change:
```
GradientGetValues/BM_ImageGradient/EbusGetValue:0/size:1024                      49.3 ms         50.0 ms           10
GradientGetValues/BM_ImageGradient/EbusGetValue:0/size:2048                       198 ms          203 ms            3
GradientGetValues/BM_ImageGradient/EbusGetValue:0/size:4096                       795 ms          797 ms            1
GradientGetValues/BM_ImageGradient/EbusGetValues:1/size:1024                     19.5 ms         19.4 ms           37
GradientGetValues/BM_ImageGradient/EbusGetValues:1/size:2048                     79.8 ms         80.4 ms            7
GradientGetValues/BM_ImageGradient/EbusGetValues:1/size:4096                      321 ms          320 ms            2
GradientGetValues/BM_ImageGradient/SamplerGetValue:2/size:1024                   70.0 ms         71.2 ms            9
GradientGetValues/BM_ImageGradient/SamplerGetValue:2/size:2048                    280 ms          281 ms            2
GradientGetValues/BM_ImageGradient/SamplerGetValue:2/size:4096                   1120 ms         1125 ms            1
GradientGetValues/BM_ImageGradient/SamplerGetValues:3/size:1024                  20.4 ms         20.7 ms           34
GradientGetValues/BM_ImageGradient/SamplerGetValues:3/size:2048                  82.4 ms         82.6 ms            7
GradientGetValues/BM_ImageGradient/SamplerGetValues:3/size:4096                   336 ms          336 ms            2
```
Signed-off-by: Chris Galvan <chgalvan@amazon.com>

### CODE REVIEW

File: Gems/GradientSignal/Code/Source/Components/ImageGradientComponent.cpp [hunk 1]
```diff
@@ -226,8 +236,19 @@ namespace GradientSignal
...
+        if (m_configuration.m_imageAsset.IsReady())
+        {
+            GetSubImageData();
+        }
+        // Otherwise for normal use-case, we queue the asset to be loaded now
+        else
+        {
+            m_imageData = AZStd::span<const uint8_t>();
+            m_configuration.m_imageAsset.QueueLoad();
+        }
```
Comment (mbalfour-amzn) [inline Gems/GradientSignal/Code/Source/Components/ImageGradientComponent.cpp:251]: Is this change needed? If the asset is already ready, the connection to the AssetBus directly above should immediately call OnAssetReady(). You might just need to move the m_imageData & QueueLoad() calls to happen _before_ the AssetBus connect instead of after.

**Review by mbalfour-amzn — APPROVED** [review]

File: Gems/GradientSignal/Code/Source/Components/ImageGradientComponent.cpp (hunk 1 above)
Comment (cgalvan) [reply Gems/GradientSignal/Code/Source/Components/ImageGradientComponent.cpp:251]: Oh nice, thanks yes that does work! Updating now :)

**Review by VickyAtAZ — APPROVED** [review]
